typescript/src/chain.ts

```diff
@@ -396,4 +430,15 @@ export interface TBTCToken {
   // TODO: Consider adding a custom type to handle conversion from ERC with 1e18
   //       precision to Bitcoin in 1e8 precision (satoshi).
   totalSupply(blockNumber?: number): Promise<BigNumber>
+
+  /**
+   * Requests redemption in one transacion using the `approveAndCall` function
+   * from the tBTC on-chain token contract. Then the tBTC token contract calls
+   * the `receiveApproval` function from the `TBTCVault` contract which burns
+   * tBTC tokens and requests redemption.
+   * @param requestRedemptionData Data required to request redemption @see
+   *        {@link RequestRedemptionData}.
+   * @returns Transaction hash of the approve and call transaction.
+   */
+  requestRedemption(requestRedemptionData: RequestRedemptionData): Promise<Hex>
```

Comment:
I would consider unwrapping parameters from the struct and passing them directly, just like we do here:
https://github.com/keep-network/tbtc-v2/blob/9249ee5ca9c71cf6754c68c5d3d891e19d77d567/typescript/src/chain.ts#L169-L174
 `RequestRedemptionData` is used only for `TBTCToken.requestRedemption` but we also have `Bridge.requestRedemption` in the same file and it is unclear why this structure is not used for both. Also, having four parameters in the function (I assume we will eliminate vault from the params) is not that bad.

Reply:
It makes sense! 💯

Reply:
06601b9